certora/specs.js

```diff
@@ -0,0 +1,51 @@
+const product = (...arrays) => arrays.reduce((a, b) => a.flatMap(ai => b.map(bi => [ai, bi].flat())));
```

Comment:
```suggestion
const product = (...arrays) => arrays.reduce((a, b) => a.flatMap(ai => b.map(bi => [ai, bi])));
```

Reply:
Ok I see it's for n > 2...

Reply:
what you propose limit the number of array to 2. The current version goes beyond:

```javascript
> product([1,2],[true,false],['a','b'])
[
  [ 1, true, 'a' ],
  [ 1, true, 'b' ],
  [ 1, false, 'a' ],
  [ 1, false, 'b' ],
  [ 2, true, 'a' ],
  [ 2, true, 'b' ],
  [ 2, false, 'a' ],
  [ 2, false, 'b' ]
]
```

Reply:
that can be usefull if you ever want to combine "specs" with "contract" with some link (like the token used)